src/BenchmarksServer/Startup.cs

```diff
@@ -560,13 +619,40 @@ private static string GetDotNetExecutable(string dotnetHome)
 
                     if (job.State == ServerState.Starting && e.Data.Contains("Application started"))
                     {
-                        job.State = ServerState.Running;
-                        job.Url = ComputeServerUrl(hostname, job.Scheme, job.Scenario);
                         Log.WriteLine($"Running job '{job.Id}' with scenario '{job.Scenario}'");
+                        job.Url = ComputeServerUrl(hostname, job.Scheme, job.Scenario);
+                        Log.WriteLine("Measuring startup time");
+                        
+                        using(var response = _httpClient.GetAsync(job.Url).GetAwaiter().GetResult())
+                        {
+                            var responseContent = response.Content.ReadAsStringAsync().GetAwaiter().GetResult();
+                            job.Startup = stopwatch.Elapsed;
+                        }
+
+                        // Measure a low-load latency by getting 5 samples synchronously
+                        // This could be done during the Client job but we are already measuring the Startup time here.
+                        const int LatencySamples = 10;
+                        var latencies = new List<TimeSpan>();
+                        for (var i = 0; i < LatencySamples; i++)
+                        {
+                            stopwatch.Restart();
+                            using(var response = _httpClient.GetAsync(job.Url).GetAwaiter().GetResult())
+                            {
+                                var responseContent = response.Content.ReadAsStringAsync().GetAwaiter().GetResult();
+                                latencies.Add(stopwatch.Elapsed);
+                            }
+                        }
+
+                        // Ignore highest and lowest values
```

Comment:
I would return all latency data from Server to Driver, and let the Driver decide how to process them and write the results to SQL.  I would definitely not discard the high and low values, since high latency should be measured rather than ignored.

In order of complexity:

1. Driver averages all data points and writes a single column to SQL.
2. Driver writes min, max, and average to SQL in 3 columns.
3. Client uses a more sophisticated latency tool with thousands of requests over many seconds, and sends back enough data to create a latency chart showing latency by percentile.